### ISSUE TITLE
how to change rendering topology

### ISSUE THREAD

**Issue by underdoeg** [liblava/liblava#26]
I am trying to render a list of vertices as points. So I have changed ```input_assembly_state.topology = VK_PRIMITIVE_TOPOLOGY_POINT_LIST``` To do so, I just made everything in the pipeline public. What would be the desired way to acomplish this?

**Comment by pezcode** [comment]
Ideally there'd be a function for setting that, but exposing every possible pipeline state without any sort of abstraction really blows up the interface. Maybe a more general way of allowing this is to make `graphics_pipeline` (and any other object with non-exposed Vulkan creation state) not have private, but protected attributes so you can easily inherit from it and then add your desired changes.

**Comment by underdoeg** [comment]
Yes. I also don't think it makes sense to wrap every feature into a function. But having access to all settings at the lowest level makes sense for a prototyping library IMO.

**Comment by TheLavaBlock** [comment]
I find both ideas quite interesting. maybe we should first switch private to protected to allow inheritance, but also offer to change these attributes from outside in the next step.

**Comment by underdoeg** [comment]
not that I am against it, but what is the advantage on changing parameters from an inherited pipeline than just creating one and changing from the outside?

**Comment by pezcode** [comment]
> one possible approach could also be to offer a simple interface as is but then have the ability to fine tune the generated pipeline config for advanced use cases. similar to the way the app is initialized.
> 
> ```c++
> auto pipeline = make_graphics_pipeline(app.device);
> pipeline.on_create = [](some_struct_with_all_vulkan_pipeline_params& params){
>       params.input_assembly_state.topology = VK_PRIMITIVE_TOPOLOGY_POINT_LIST;
> }
> ```

I actually really like that approach 👌
